pennylane/transforms/core/transform.py

```diff
@@ -177,6 +178,47 @@ def qnode_circuit(a):
         >>> function1(function2(result))
         np.float64(0.499...)
 
+    .. details::
+        :title: Setup inputs
+
+        The ``setup_inputs`` function will independently applied prior to any application of
+        the transform. This allows for validation of the inputs, separation into positional and
+        keyword arguments, and specification of a call signature and docstring for transforms
+        without a tape definition.
+
+        .. code-block:: python
+
+            def setup_inputs(a, b=1, metadata : str = "my_value"):
+                "Docstring for my_transform."
+                return (a, b), {"metadata": metadata}
+
+            my_transform = qml.transform(pass_name="my_pass", setup_inputs=setup_inputs)
+
+            @qml.qnode(qml.device('default.qubit', wires=4))
+            def circuit():
+                return qml.expval(qml.Z(0))
+
+        This allows us to perform eager input validation and set default values.
+
+        >>> my_transform(circuit)
+        Traceback (most recent call last):
+            ...
+        TypeError: setup_inputs() missing 1 required positional argument: 'a'
```

Comment:
Would it be possible for the error message to tell us exactly which transform is getting incorrect arguments? This would make debugging easier when we have multiple transforms applied on the QNode.